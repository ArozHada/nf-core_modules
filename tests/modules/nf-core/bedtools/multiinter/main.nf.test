nextflow_process {

    name "Test Process BEDTOOLS_MULTIINTER"
    script "modules/nf-core/bedtools/multiinter/main.nf"
    process "BEDTOOLS_MULTIINTER"
    config "tests/modules/nf-core/bedtools/multiinter/nextflow.config"
    tag "bedtools_multiinter"

    test("Sarscov2 Genome BED") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            [
                              file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true),
                              file(params.test_data['sarscov2']['genome']['test2_bed'], checkIfExists: true)
                            ]
                           ]
                input[1] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("bed") }
            )
        }
    }

    test("Sarscov2 Genome BED With Sizes") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            [
                              file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true),
                              file(params.test_data['sarscov2']['genome']['test2_bed'], checkIfExists: true)
                            ]
                           ]
                input[1] = file(params.test_data['sarscov2']['genome']['genome_sizes'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("bed_sizes") }
            )
        }
    }
}
