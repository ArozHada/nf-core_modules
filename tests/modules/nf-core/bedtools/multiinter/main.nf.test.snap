{
    "bed": {
        "content": [
            [
                "test.bed:md5,03d8d889a19cf26e038868775bbcbaa7",
                "versions.yml:md5,be7d8cd702693d431094d8dfee085108"
            ]
        ],
        "timestamp": "2023-04-11T22:13:57+0000"
    },
    "bed_sizes": {
        "content": [
            [
                "test.bed:md5,03d8d889a19cf26e038868775bbcbaa7",
                "versions.yml:md5,be7d8cd702693d431094d8dfee085108"
            ]
        ],
        "timestamp": "2023-04-11T22:13:57+0000"
    }
}